dashboards-observability/public/components/common/search/autocomplete_logic.ts

```diff
@@ -157,6 +186,7 @@ const filterSuggestions = (suggestions: AutocompleteItem[], prefix: string) => {
 
 // Main logic behind autocomplete (Based on most recent inputs)
 export const getFullSuggestions = async (
+  base: string,
```

Comment:
how does base change logic of autocomplete?

Reply:
It does not, refer to this [comment](https://github.com/opensearch-project/observability/pull/480#discussion_r811399199)

Reply:
why not trigger it like `getSuggestions(baseQuery + query, dslService);`, and remove `base` in both get suggestions?

Reply:
I use `query` to fill the suggestions because the `baseQuery` should not be part of the object which populates the search bar when a suggestion is selected.

Reply:
hm i see, i would probably do 
```js
getSuggestions={appLogEvents ? getSuggestionsAfterSource : (base, curr, dslService) => getFullSuggestions(curr, dslService)}
```
but it feels more ugly. yours is ok i think.

Are the logic from `getFullSuggestions` reusable in `getSuggestionsAfterSource`?

Reply:
It is very similar but the big difference is that `getFullSuggestions` has to get the words one at a time while `getSuggestionsAfterSource` can get a pre-existing query and parse it and continue with the autocomplete. I think we can just use `getSuggestionsAfterSource` instead of `getFullSuggestions` everywhere else later if we also add suggesting for the source.